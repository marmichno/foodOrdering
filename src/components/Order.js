import {Carousel} from './common/Carousel';
import {useState, useEffect} from 'react';
import {motion, AnimatePresence} from 'framer-motion';

const containerVariants = {
    hidden: {
        x: -2000
    },
    visible: {
        x: 0,
        transition: {
            type: 'spring',
            delay: 0.5,
            mass: 1,
            damping: 12
        }
    },
    exit: {
        x:'-100vw',
        transition: {ease: 'easeInOut', duration:0.5}
    }
  }

export const Order = () => {

    const [currentActive, setCurrentActive] = useState(null);
    const foodTypes = ['Sushi rolls', 'Ramen', 'Prawns', 'Sets'];
    const [choosenFood, setChoosenFood] = useState('Sushi rolls');

    useEffect(() =>{
        const allHeaders = document.querySelectorAll('.foodHeadersContainer h1');
        allHeaders.forEach(element => element.classList.add('notActive'));
        allHeaders[0].classList.add('active');
        setCurrentActive(allHeaders[0].innerHTML);
    }, []);

    const changeActive = (e) => {
        const allHeaders = document.querySelectorAll('.foodHeadersContainer h1');
        allHeaders.forEach(element => element.classList.remove('active'));
        e.target.classList.add('active');
        setCurrentActive(e.target.innerHTML);
        setChoosenFood(e.target.innerHTML);
    }

    return(
        <div className="orderMainContainer">
            <motion.div
            variants={containerVariants}
            initial="hidden"
            animate="visible"
            exit="exit"
            className="foodHeadersContainer">
                {foodTypes.map(value => {
                    return <h1 onClick={changeActive}>{value}</h1>
                })}
            </motion.div>

            <Carousel choosenFood={choosenFood}/>

    </div>
    )
}