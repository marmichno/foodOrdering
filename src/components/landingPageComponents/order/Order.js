import {Products} from '../products/Products';
import {ProductsGroups} from '../productsGroups/ProductsGroups';
import {CgScrollV} from 'react-icons/cg';
import {motion} from 'framer-motion';


const scrollContainerVariants = {
    hidden: {
        y: 500
    },
    visible: {
        y: 0,
        transition: {
            type: 'spring',
            delay: 0.5,
            mass: 1,
            damping: 12
        }
    },
    exit: {
        y:500,
        transition: {ease: 'easeInOut', duration:0.5}
    }
}

const bouncyAnimation = {
    visible: {
        scale: 1.1,
        transition: {
            duration:0.3,
            yoyo:Infinity
        }
    }
}

export const Order = () => {

    return(
        <div className="orderMainContainer" data-testid="orderContainer">
            
            <ProductsGroups/>
            <Products/>

            <motion.div
            variants={scrollContainerVariants}
            initial="hidden"
            animate="visible"
            exit="exit"
            className="scroll">
                <motion.div
                variants={bouncyAnimation}
                animate="visible" 
                className="icon">
                    <CgScrollV/>
                </motion.div>
            </motion.div>
    </div>
    )
}

export default Order;
