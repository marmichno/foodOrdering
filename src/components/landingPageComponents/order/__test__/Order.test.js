import {Order} from '../Order';
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

const store = createStore(() => [], {}, applyMiddleware());

test("matches snapshot", () => {
    const tree = TestRenderer.create(<Provider store={store}><Order/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});