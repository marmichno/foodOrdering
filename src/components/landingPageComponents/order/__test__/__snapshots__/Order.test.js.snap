// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="orderMainContainer"
  data-testid="orderContainer"
>
  <div
    className="foodHeadersContainer"
    style={
      Object {
        "transform": "translateX(-2000px) translateZ(0)",
      }
    }
  >
    <div
      className="foodGroupsContainer"
    >
      <div
        className="list"
      >
        <div
          className="selected"
        >
          <h1
            data-testid="selected"
          />
          <hr />
        </div>
        <div
          className="toggle"
          data-testid="dropDown"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 48C141.125 48 48 141.125 48 256s93.125 208 208 208 208-93.125 208-208S370.875 48 256 48zm0 272l-96-96h192l-96 96z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="foodMainContainer"
    style={
      Object {
        "transform": "translateX(2000px) translateZ(0)",
      }
    }
  />
  <div
    className="scroll"
    style={
      Object {
        "transform": "translateY(500px) translateZ(0)",
      }
    }
  >
    <div
      className="icon"
      style={Object {}}
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.17154 16.8182L7.75732 18.2324L12 22.475L16.2426 18.2324L14.8284 16.8182L12 19.6466L9.17154 16.8182Z"
          fill="currentColor"
        />
        <path
          d="M14.8284 7.182L16.2426 5.76779L12 1.52515L7.75733 5.76779L9.17155 7.182L12 4.35357L14.8284 7.182Z"
          fill="currentColor"
        />
        <path
          clipRule="evenodd"
          d="M12 9.00018C13.6569 9.00018 15 10.3433 15 12.0002C15 13.657 13.6569 15.0002 12 15.0002C10.3431 15.0002 9 13.657 9 12.0002C9 10.3433 10.3431 9.00018 12 9.00018ZM12 11.0002C12.5523 11.0002 13 11.4479 13 12.0002C13 12.5525 12.5523 13.0002 12 13.0002C11.4477 13.0002 11 12.5525 11 12.0002C11 11.4479 11.4477 11.0002 12 11.0002Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;
